File: backend/Portal.Domain/Entities/Turma.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Portal.Domain.Entities
{
    public class Turma: BaseEntity
    {
        [Required]
        public string Name { get; set; }
        [Required]
        public Guid EscolaId { get; set; }
        [Required]
        public DateTime CreatedDate { get; set; }
        public Escola Escola { get; set; }
        public Turma()
        {
            CreatedDate = DateTime.UtcNow;
        }
    }
}

File: frontend/src/app/turma/turma.ts
import { Escola } from "../escola/escola";

export interface ITurmaRequest {
    id: number;
    escolaId: number;
    name: string;
    escola: Escola;
    createdDate: Date;
}

export interface ITurmaResponse {
    name: string;
    escolaId: number;
}

export interface ITurmaGrid {
    name: string;
    nameEscola: string;
    createdDate: Date;
}


